ChangeLog.md

```diff
@@ -0,0 +1,7 @@
+# Mbed TLS change history
+
+## Unreleased changes
+
+## Older changes
+
+Please see the [`ChangeLog`](ChangeLog) file.
```

Comment:
Should we also rename the old ChangeLog to something clearer (e.g. ChangeLog_Old) to avoid confusion with ChangeLog.md?

Reply:
My thinking was that I'd rather keep `ChangeLog` under this name until we're ready to switch, so that we can keep merging PRs that modify it.

But on second thoughts git will resolve a **rename**/modify conflict. What it wouldn't resolve is delete/modify.

And on third thoughts, `ChangeLog.md` is not actually doing anything useful now, except for manual testing of `assemble_changelog.py`. So I'm going to remove `ChangeLog.md`, which will reduce the confusion.